### PR TITLE
Upstream aurora-postgres-resources

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Upstream aurora-postgres-resources

## why
* Clients can take advantage of upstream fixes

## references
N/A

### CODE REVIEW

File: modules/aurora-postgres-resources/default.auto.tfvars [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# This file is included by default in terraform plans
+
+enabled = false
+
+name = "pg"
```
Comment (aknysh) [inline modules/aurora-postgres-resources/default.auto.tfvars:5]: let's move `name` to YAML config (we had issues in Spacelift when we define vars in many places)

File: modules/aurora-postgres-resources/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        environment_secondary: ue2
+        engine: aurora-postgresql
+        engine_version: "12.4"
+        engine_mode: global
+        cluster_family: aurora-postgresql12
+        cluster_size: 1
+        instance_type: db.r4.large
+        iam_database_authentication_enabled: true
+        storage_encrypted: true
+        deletion_protection: false
```
Comment (nitrocode) [inline modules/aurora-postgres-resources/README.md:28]:
```suggestion
        aurora_postgres_component_name: aurora-postgres-example
        additional_users:
          example:
            db_user: example
            db_password: ""
            grants:
              - grant: [ "ALL" ]
                db: example
                object_type: database
                schema: null
```

**Review by aknysh — COMMENTED** [review]
please see comments

File: modules/aurora-postgres-resources/default.auto.tfvars (hunk 1 above)
Comment (nitrocode) [inline modules/aurora-postgres-resources/default.auto.tfvars:5]:
```suggestion

name = "pg"
```

File: modules/aurora-postgres-resources/providers.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+variable "import_role_arn" {
+  type        = string
+  default     = null
+  description = "IAM Role ARN to use when importing a resource"
+}
+
+provider "postgresql" {
+  # host      = local.cluster_endpoint
```
Comment (nitrocode) [inline modules/aurora-postgres-resources/providers.tf:31]:
```suggestion
  host      = local.cluster_endpoint
```

File: modules/aurora-postgres-resources/README.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+| Name | Description |
+|------|-------------|
+| <a name="output_additional_databases"></a> [additional\_databases](#output\_additional\_databases) | Additional databases |
+| <a name="output_additional_users"></a> [additional\_users](#output\_additional\_users) | Additional users |
+| <a name="output_read_only_users"></a> [read\_only\_users](#output\_read\_only\_users) | Read-only users |
+<!-- END OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
+
+
+## References
+* [cloudposse/terraform-aws-components](https://github.com/cloudposse/terraform-aws-components/tree/master/modules/TODO) - Cloud Posse's upstream component
```
Comment (nitrocode) [inline modules/aurora-postgres-resources/README.md:104]:
```suggestion
* [cloudposse/terraform-aws-components](https://github.com/cloudposse/terraform-aws-components/tree/master/modules/aurora-postgres-resources) - Cloud Posse's upstream component
```

**Review by aknysh — APPROVED** [review]
we prob need to write the auto-generated DB passwords for all additional users into SSM, but let's do it in a separate PR

**Pull request merged**